beacon-chain/graffiti/graffiti-proposal-brief.md

```diff
@@ -0,0 +1,267 @@
+# Proposal: Automatic Client Version Graffiti (Brief)
+
+## TL;DR
+
+Add automatic EL+CL version info to block graffiti following [ethereum/execution-apis#517](https://github.com/ethereum/execution-apis/pull/517). Format: `GE168dPR63af` (Geth + Prysm). User graffiti always takes precedence.
+
+## Problem
+
+**Prysm hasn't yet implemented the ecosystem standard.** Prysm currently leaves graffiti empty (when no user graffiti is configured), providing no on-chain visibility into client versions.
+
+**Why it matters**: On-chain visibility of client diversity is much easier with standardized graffiti.
+
+## Proposed Solution
+
+### Format
+```
+<EL_CODE><EL_COMMIT><CL_CODE><CL_COMMIT>
+GE      168d      PR      63af       = "GE168dPR63af"
+```
+
+- **EL_CODE**: 2-letter code (GE=Geth, NM=Nethermind, BU=Besu, RH=Reth, EG=Erigon)
+- **EL_COMMIT**: First 4 hex chars of execution client's git commit (from `engine_getClientVersionV1` JSON-RPC response)
+- **CL_CODE**: PR (Prysm)
+- **CL_COMMIT**: First 4 hex chars of Prysm's git commit (from build-time ldflags via `version.GetCommitPrefix()`)
+
+### Priority Order
+```
+1. User graffiti (VC flag, proposer settings, graffiti file)  ← Always wins
+   ↓
+2. Auto-generated "GE168dPR63af"                              ← New feature
+   ↓
+3. Default "Prysm/v6.1.0"                                     ← Fallback
+```
+
+**Key principle**: User control preserved. No opt-out flag needed.
+
+### Examples
+
+| Configuration | Result                             |
+|--------------|------------------------------------|
+| No graffiti | `GE168dPR63af` ✨ **NEW**           |
+| `--graffiti "🚀"` | `🚀` (unchanged)                   |
+| Proposer settings | Custom graffiti (unchanged)        |
+| Old Geth (no API) | `Prysm/v6.1.0` (graceful fallback) |
```

Comment:
Can we follow the "[flexible standard](https://hackmd.io/@wmoBhF17RAOH2NZ5bNXJVg/BJX2c9gja)" that was proposed alongside the engine API PR? These examples imply that we would only set the custom graffiti if no user graffiti is specified.

If we do that it could also be nice to emit a log to warn the user that they are using up valuable graffiti real estate that could be used for client diversity. For example look at [this block](https://beaconcha.in/slot/12968727): `Twinstake for EtherFi`. Maybe they would be happy to just set the flag to `EtherFi` if they knew it helped client diversity analysis. As an aside, it's a little pointless for the big stakers to set custom graffiti since the chain explorers know who they are anyway.